test/integration/transaction_list_ft_tests.js

```diff
@@ -598,6 +599,36 @@ describe('The /transactions endpoint filtering by states)', () => {
       })
       .end(done)
   })
+
+  it('should correctly map cardholder_name and last_digits_card_number to the CSV download link', function (done) {
+    connectorMock.get(CONNECTOR_ALL_CARD_TYPES_API_PATH)
+      .reply(200, ALL_CARD_TYPES)
+
+    const expectedCSVFilters = {
+      cardholderName: 'example-name',
+      lastDigitsCardNumber: '9304'
+    }
+
+    const connectorData = {
+      'results': []
+    }
+
+    connectorMockResponds(200, connectorData, expectedCSVFilters)
+
+    request(app)
+      .get(paths.transactions.index)
+      .query(expectedCSVFilters)
+      .set('Accept', 'application/json')
```

Comment:
We are trying to avoid testing the "app/integration" tests and setting the `application/json` as it returns JSON and not actual page, but looks like all of the download CSV tests are implemented this way.
We have some Cypress tests for transaction list here https://github.com/alphagov/pay-selfservice/blob/5652dff08d64139f15e221051ee643b8ff844795/test/cypress/integration/user/transaction_search_spec.js
but it is not used for CSV download yet.
Given the urgency of the fix, I think we can refactor / add Cypress testing at some point later

Reply:
I agree with this - we should definitely add these tests following the potential upcoming updates to Cypress testing.